lib/services/services_linux.c

```diff
@@ -196,6 +196,40 @@ add_action_env_vars(const svc_action_t *op)
     }
 }
 
+static void
+pipe_in_single_parameter(gpointer key, gpointer value, gpointer user_data)
+{
+    char buffer[500];
+    svc_action_t *op = user_data;
+    int ret, total = 0, len;
+
+
+    len = snprintf(buffer, sizeof(buffer), "%s=%s\n", (char *)key, (char *) value);
```

Comment:
Otherwise you have to take care if dynamic allocation went OK.
Actually I was thinking if it would be worthwhile adding the fd to the mainloop as not to be limited by the file-buffer-size ... then we would probably print everything in a large buffer first and memorize that inside svc_action.

Reply:
OK. FYI crm_strdup_printf() does an assert for non-null, so we don't have to worry about checking the result.